lib/mail/body.rb

```diff
@@ -154,9 +154,8 @@ def encoded(transfer_encoding = '8bit')
         encoded_parts = parts.map { |p| p.encoded }
         ([preamble] + encoded_parts).join(crlf_boundary) + end_boundary + epilogue.to_s
       else
-        be = get_best_encoding(transfer_encoding)
```

Comment:
if be is different than desired transfer_encoding, the content is encoded, but the transfer encoding stays the same